src/collectors/collector_worker.go

```diff
@@ -24,7 +25,8 @@ type CollectorWorker struct {
 func (cw *CollectorWorker) RunCollectors() {
 	cw.registerCollectors()
 	// Create a channel with a maximum size of configured concurrency
-	queue := make(chan bool, cw.RunConfig.Concurrency)
+	queue := make(chan struct{}, cw.RunConfig.Concurrency)
```

Comment:
Is the queue still necessary with the WaitGroup?

Reply:
Yes, queue is still used to limit concurrency. (line 41 blocks until there is an available slot)

Reply:
Ah, yes, I see now thanks.

This is a pretty good article that shows limited concurrency doing exactly what you've got here. ;) https://pocketgophers.com/limit-concurrent-use/